torch/fx/experimental/symbolic_shapes.py

```diff
@@ -3696,6 +3696,10 @@ def _maybe_evaluate_static(
         new_shape_env = {}
         new_range_env = {}
         for idx, k in enumerate(symbols):
+            if isinstance(self.var_to_val.get(k, None), SingletonInt):
```

Comment:
@ezyang should probably take a look?

Reply:
this LGTM. I didn't check the rest of the PR